test/inductor/test_torchinductor.py

```diff
@@ -3646,14 +3646,37 @@ def fn(a, b):
             )
 
         for ind_dtype in (torch.int32, torch.int64):
-            self.common(
-                fn,
-                (
-                    torch.randn(8, 8, 8),
-                    torch.tensor([0, 0, 2, 1], dtype=ind_dtype),
-                ),
+            for inp_dtype in (torch.float32, torch.half, torch.bfloat16):
+                self.common(
+                    fn,
+                    (
+                        torch.randn(8, 8, 8, dtype=inp_dtype),
+                        torch.tensor([0, 0, 2, 1], dtype=ind_dtype),
+                    ),
+                )
+
+    def test_index_add(self):
+        def fn(a, b, c):
+            return (
+                torch.index_add(a, 0, b, c, alpha=1),
+                torch.index_add(a, 1, b, c, alpha=-1),
             )
 
+        for ind_dtype in (torch.int64, torch.int64):
+            for inp_dtype in (torch.float32,):
+                # torch.bfloat16, torch.half
+                # disable low precision since non-deterministic parallel reduction have larger impact
+                # and also suffered from not using accumulate data type
+                # https://github.com/pytorch/pytorch/issues/93582
```

Comment:
This test is shared by both CPU and GPU. Consider to disable other dtypes only for CPU?